src/Orleans/Streams/Predicates/IStreamNamespacePredicate.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Orleans.Streams
+{
+    public interface IStreamNamespacePredicate
+    {
+        bool IsMatch(string streamNamespace);
```

Comment:
@jason-bragg The only problem with this that it limits the cachebility of the subscribers in `ImplicitStreamSubscriberTable` since you have to recalculate that list per stream ID + namespace + provider name. With stream IDs having dynamic nature, I guess, it might have significant impact.

What do you think the performance penalty will be in this case? If it is insignificant, I will be happy to make this change.